### ISSUE TITLE
The UI does not respond after testing the project (Twine 2.3.0 for Windows)

### ISSUE THREAD

**Issue by Flannastri** [klembot/twinejs#548]
I have a strange issue with Twine 2.3.0 for Windows. As soon as I test the project using the «Play» button, adding new passages with the «+Passage» button just stop working. It works only once, but the new passage replaces the starting one (although its content remains the same). And after that the button stops responding. I also cannot move passages, some UI elements are not responding either (can't press «Play» again, for example).

First, I made two simple links:
![image](https://user-images.githubusercontent.com/49790551/56432041-aad22b80-62d4-11e9-8f84-db544a96cbcd.png)

Then check it with the «Play» button. Then I add a new passage and see this:
![image](https://user-images.githubusercontent.com/49790551/56432072-c6d5cd00-62d4-11e9-9eea-b8cd309e912f.png)

This «Untitled passage» actually contains the text of previous starting passage, which he replaced:
![image](https://user-images.githubusercontent.com/49790551/56432235-46639c00-62d5-11e9-8ffa-259e420453e0.png)
...but links in it just don't work, as seen in the previous image. And this passage kinda «stuck» in place, it cannot be moved.

P.S. Sorry for my English, I hope I explained correctly. :)

**Comment by klembot** [comment]
I'm sorry, I can't reproduce this (at least on my Mac). For starters, could you try temporarily removing all story files from your library on disk and try from a completely new setup, to see if it happens again?

If it does, could you also try opening the developer console (Toggle Developer Tools under the View menu) and see if any errors appear there?

**Comment by Flannastri** [comment]
Tried to do as you suggest in new 2.3.1 version — same story, it happens all the time.
And yes, the dev console shows some errors. Not sure that this is what you need, but here it is.

```
Uncaught (in promise) DOMException: Failed to execute  twine.js:6 
'insertBefore' on 'Node': The node before which the new node is to be inserted is not a child of this node.
    at file:///C:/Program%20Files/Twine/resources/app.asar/dist/web-electron/twine.js:6:176722
    at oi (file:///C:/Program%20Files/Twine/resources/app.asar/dist/web-electron/twine.js:6:176765)
    at file:///C:/Program%20Files/Twine/resources/app.asar/dist/web-electron/twine.js:6:167256
    at file:///C:/Program%20Files/Twine/resources/app.asar/dist/web-electron/twine.js:6:167384
    at file:///C:/Program%20Files/Twine/resources/app.asar/dist/web-electron/twine.js:6:167059
    at Wr (file:///C:/Program%20Files/Twine/resources/app.asar/dist/web-electron/twine.js:6:167065)
    at file:///C:/Program%20Files/Twine/resources/app.asar/dist/web-electron/twine.js:6:171184
    at ka.setOption (file:///C:/Program%20Files/Twine/resources/app.asar/dist/web-electron/twine.js:6:262049)
    at file:///C:/Program%20Files/Twine/resources/app.asar/dist/web-electron/twine.js:29:39804

Uncaught TypeError: Cannot read property format.js?callback=storyFormat:1 'length' of undefined
    at format.js?callback=storyFormat:1
    at Array.forEach (<anonymous>)
    at a (format.js?callback=storyFormat:1)
    at twine.js:6
    at twine.js:6
    at twine.js:6
    at Wr (twine.js:6)
    at HTMLTextAreaElement.<anonymous> (twine.js:6)

Uncaught (in promise) MouseEvent index.html#!/stories/2584350:1

Uncaught (in promise) DOMException: Failed to execute  twine.js:6  
'insertBefore' on 'Node': The node before which the new node is to be inserted is not a child of this node.
    at file:///C:/Program%20Files/Twine/resources/app.asar/dist/web-electron/twine.js:6:176722
    at oi (file:///C:/Program%20Files/Twine/resources/app.asar/dist/web-electron/twine.js:6:176765)
    at file:///C:/Program%20Files/Twine/resources/app.asar/dist/web-electron/twine.js:6:167256
    at file:///C:/Program%20Files/Twine/resources/app.asar/dist/web-electron/twine.js:6:167384
    at file:///C:/Program%20Files/Twine/resources/app.asar/dist/web-electron/twine.js:6:167059
    at Wr (file:///C:/Program%20Files/Twine/resources/app.asar/dist/web-electron/twine.js:6:167065)
    at file:///C:/Program%20Files/Twine/resources/app.asar/dist/web-electron/twine.js:6:171184
    at ka.setOption (file:///C:/Program%20Files/Twine/resources/app.asar/dist/web-electron/twine.js:6:262049)
    at file:///C:/Program%20Files/Twine/resources/app.asar/dist/web-electron/twine.js:29:39804

Uncaught (in promise) MouseEvent index.html#!/stories/2584350:1 

Uncaught (in promise) DOMException: Failed to execute   twine.js:6 
'insertBefore' on 'Node': The node before which the new node is to be inserted is not a child of this node.
    at file:///C:/Program%20Files/Twine/resources/app.asar/dist/web-electron/twine.js:6:176722
    at oi (file:///C:/Program%20Files/Twine/resources/app.asar/dist/web-electron/twine.js:6:176765)
    at file:///C:/Program%20Files/Twine/resources/app.asar/dist/web-electron/twine.js:6:167256
    at file:///C:/Program%20Files/Twine/resources/app.asar/dist/web-electron/twine.js:6:167384
    at file:///C:/Program%20Files/Twine/resources/app.asar/dist/web-electron/twine.js:6:167059
    at Wr (file:///C:/Program%20Files/Twine/resources/app.asar/dist/web-electron/twine.js:6:167065)
    at file:///C:/Program%20Files/Twine/resources/app.asar/dist/web-electron/twine.js:6:171184
    at ka.setOption (file:///C:/Program%20Files/Twine/resources/app.asar/dist/web-electron/twine.js:6:262049)
    at file:///C:/Program%20Files/Twine/resources/app.asar/dist/web-electron/twine.js:29:39804

**Comment by gwsk5597** [comment]
I'd just like to add that I've experienced this issue as well. It's very much a real problem.

**Comment by Kln95130** [comment]
Same here. The only "fix" that I have found is to restart the computer.

**Comment by potatocubed** [comment]
Me too.

I add a new passage, and it 'vanishes' the one I was working on and sticks a new, unclickable passage in the middle of the field.

If I back out to the main menu and come back in then the new passage is manipulable as normal and my old passage is back.

If I reproduce the lock-up I can hover my mouse over the new passage and delete it -- whereupon it deletes the passage I was last working on.

**Comment by Matt7477** [comment]
This is an issue that I am experiencing on my Mac, and a number of the students I teach (those who updated to 2.3.x) are also experiencing issues.

It seems as though the program is not detecting a new passage has been placed in the story. The only way I can temporarily fix it is to close out the program and restart it.

I don't seem to have an issue using the web browser version.

My specs are...

Twine 2.3.1 locally on my computer.
Computer software: macOS Mojave version 10.14.4

**Comment by Kazekai** [comment]
Same issue. Creating new passages just overwrites the last one for me after a while, the 'play' button refuses to work or it displays pages with a blank screen, and the only fix is to restart Twine, although that won't fix it for long.

**Comment by klembot** [comment]
I am still having a lot of trouble duplicating this bug myself. I know that it's frustrating on everyone else's end. What would help me out here is if anyone can come up with a concrete set of steps that triggers this bug from a fresh start.

What might also help here is if you see a blank screen when playing a story, open the developer tools (under the View menu). This may display an error message. This would also help me start to track down the problem.

**Comment by lazybibowl** [comment]
I guess I have the same problem as Kazekai.

![1](https://user-images.githubusercontent.com/50878645/58108182-8dff7f80-7c15-11e9-8370-dbba890637f0.PNG)

**Comment by Flannastri** [comment]
> What would help me out here is if anyone can come up with a concrete set of steps

1. Well, first I create a new story. In my case, this is the first launch in the newly installed client, in which I have not worked yet. But this is true for any story that is created in Twine 2.3.0 or Twine 2.3.1.
For the time elapsed since the first post, I tried Twine 2.3.1 on another computer (also Windows x64) – the same thing.
2. Then I create a couple of passages so that there are several steps to check. The important point is that at this stage I can still create as many passages as I want and generally work normally.
3. Then I play the story with the Play button.
4. Then I try to create a new passage – it replaces the starting one and here is the bug.
5. If not immediately, then on the next attempt. I did not understand a clear connection here. Usually this is the first passage after playing the story, but not always. If not the first, then the second. It will happen anyway.
6. Passages cannot be moved with the cursor, the «New passage» button stops working, and then other interface elements (for example, Play) stop working too.

**Comment by klembot** [comment]
This is the frustrating part-- these steps work for me. This is exactly what I am doing. I have Snowman 1.3.0 set as my default format.

1. I create a new story.
2. I click the green + Passage button five times.
3. I click the Play button. The story opens as usual.
4. I click the green + Passage button in the original window. This works fine.
5. I try editing a passage. This also works fine.

**Comment by Kln95130** [comment]
Have you tried with Harlowe 3? After switching a story from Harlowe 2 to Harlowe 3, I reproduced this bug, not even having to it play. I edited a passage, created another, then tried to create another. Here is the error messages I have in my console. The first one repeats each time I try to open a passage:

![image](https://user-images.githubusercontent.com/8339550/58372053-63b10900-7f18-11e9-8e42-52e4df355897.png)

**Comment by klembot** [comment]
I tried just now with Harlowe 3.0.2, and I get those errors, but it does not seem to interfere with my ability to continue editing the story.

**Comment by Kln95130** [comment]
I may have another clue: the bug does not appear if I let the default name of the passages (Passage sans titre, Passage sans titre 1, etc.)
However, it triggered again, as soon as I renamed one of the passage with "2"

**Comment by klembot** [comment]
OK, now I can reproduce this bug. Thank you for your help!

**Comment by Kln95130** [comment]
You're welcome :-)

**Comment by Flannastri** [comment]
Yeah, this is the very thing that I did not take into account. As soon as you give a passage a new name instead of the standard one, a bug appears. Does not depend on the format, apparently. Bug appeared on Harlowe, SugarCube and SnowMan for me. I apologize for not having noticed this before.
Kln95130, thank you a lot. :3
